content/docs/user-guide/dvc-files-and-directories.md

```diff
@@ -138,6 +162,56 @@ the possible following fields:
 
 `dvc.yaml` files also support `# comments`.
 
+### dvc.lock file
```

Comment:
Should user commit lockfile to the VCS?

Reply:
It's a good note but I'm not sure we want to talk about that in this doc. We would need similar notes on all the other types of docs too. WDYT @shcheklein ?

Reply:
@skshetry please, give some suggestion on this?

Reply:
> You should commit the `dvc.lock` file to your project repo so that all team-members can reuse the outputs of your experiment without re-running.

Reply:
Note that we might need to explain difference with `run-cache` later and might need to compare these two. 

1. Run-Cache can generate outputs in a suitable condition and that too indirectly (eg: only on `run/repro`), whereas a lockfile will always provide you with outputs without rerunning.
2. RunCache can generate lockfile. Even though you have a run-cache, you should still use lockfile because for the outputs. 
3. `get`/`import` commands work well only if you have a lockfiile.

Reply:
But, this can be added later when documenting run-cache.

Reply:
Interesting. Is there a run-cache issue we can add all these comments to? Even if it's in the core repo.

Reply:
Moved to https://github.com/iterative/dvc.org/issues/1289#issuecomment-648595194